apps/server-asset-sg/src/features/asset-edit/asset-edit.repo.ts

```diff
@@ -320,6 +323,45 @@ export class AssetEditRepo implements Repo<AssetEditDetail, number, AssetEditDat
     }
     return detailResult.right as AssetEditDetail;
   }
+
+  private async validateReferencesOrThrow(data: AssetEditData, id?: number): Promise<void> {
+    // check if any of the siblings are in another workgroup
+    for (const assetYId of data.patch.siblingAssetIds) {
+      const siblingCandidate = await this.prismaService.asset.findUnique({
+        where: { assetId: assetYId },
+        select: { assetId: true, workgroupId: true },
+      });
+      if (siblingCandidate?.workgroupId !== data.patch.workgroupId) {
+        throw new Error('Sibling assets must be in the same workgroup as the edited asset');
+      }
+    }
+
+    // check if the parent asset is in another workgroup
+    const assetMainId = O.toUndefined(data.patch.assetMainId);
+    if (assetMainId) {
+      const assetMain = await this.prismaService.asset.findUnique({
+        where: { assetId: assetMainId },
+        select: { assetId: true, workgroupId: true, titlePublic: true },
+      });
+      if (assetMain?.workgroupId !== data.patch.workgroupId) {
+        throw new Error('Cannot assign parent asset from different workgroup');
+      }
+    }
+
+    // check if any of the subordinate assets are in another workgroup for exisiting assets
+    if (id) {
+      const childAssets = await this.prismaService.asset.findMany({
+        where: { assetMainId: id },
+        select: { assetId: true, workgroupId: true, titlePublic: true },
+      });
+
+      for (const child of childAssets) {
+        if (child.workgroupId !== data.patch.workgroupId) {
+          throw new Error('The subordinate assets must be in the same workgroups as the parent asset');
```

Comment:
`subordinate` means something different than `child`.
```suggestion
          throw new Error('Child assets must be in the same workgroup as the parent asset');
```